### PR TITLE
Added two DTOs, and updated subscription plan to use service provider…

### PR DESCRIPTION
**Pull request by w280749941**
… rather than subscription id

### CODE REVIEW

File: SubscriptionManagementSystem/persistent-common/src/main/java/com/subsmanagement/entity/SubscriptionPlan.java [hunk 1]
```diff
@@ -17,5 +17,5 @@
     private BigDecimal cost;
     private LocalDateTime createTime;
     private LocalDateTime lastUpdateTime;
-    private Integer subscriptionId;
+    private Integer serviceProviderId;
```
Comment (xiao-peng1006) [inline SubscriptionManagementSystem/persistent-common/src/main/java/com/subsmanagement/entity/SubscriptionPlan.java:20]: subscriptions under the same service provider may have different subscription plans to choose?

**Review by xiao-peng1006 — COMMENTED** [review]
changes on the subscription plan may not be accurate

File: SubscriptionManagementSystem/persistent-common/src/main/java/com/subsmanagement/entity/SubscriptionPlan.java (hunk 1 above)
Comment (w280749941) [reply SubscriptionManagementSystem/persistent-common/src/main/java/com/subsmanagement/entity/SubscriptionPlan.java:20]: Yes. A service provider should provide multiple plans.

**Review by xiao-peng1006 — APPROVED** [review]
Reviewed and approved

**Pull request merged**